data_model/src/events/data/filters.rs

```diff
@@ -95,10 +95,15 @@ impl Filter for EntityFilter {
 }
 
 #[derive(Clone, PartialOrd, Ord, Eq, Debug, Decode, Encode, Serialize, IntoSchema)]
-/// Filter that accepts an data event whose origin matches what this filter specifies.
-pub struct OriginFilter<T: Origin>(<T::Origin as Identifiable>::Id);
+/// Filter that accepts a data event with the matching origin.
+pub struct OriginFilter<T: HasOrigin>(<T::Origin as Identifiable>::Id)
+where
+    <T::Origin as Identifiable>::Id: IntoSchema;
```

Comment:
This where clause may be introduced, but I don't see the value of leaving room for not implementing `IntoSchema` for `Id`

Reply:
https://github.com/hyperledger/iroha/pull/2497#discussion_r923494428